source/common/grpc/common.cc

```diff
@@ -71,9 +86,13 @@ Http::MessagePtr Common::prepareHeaders(const std::string& upstream_cluster,
   return message;
 }
 
-void Common::checkForHeaderOnlyError(Http::Message& http_response) {
+void Common::validateHeaders(const Http::HeaderMap& headers) {
+  if (Http::Utility::getResponseStatus(headers) != enumToInt(Http::Code::OK)) {
+    throw Exception(Optional<uint64_t>(), "non-200 response code");
```

Comment:
This was here before. I'm actually going to undo my changes to this file, since I've inlined the relevant bits into `async_client_impl.h`.